flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/exceptionhistory/RootExceptionHistoryEntry.java

```diff
@@ -36,57 +36,91 @@
 
     private static final long serialVersionUID = -7647332765867297434L;
 
-    private final Collection<ExceptionHistoryEntry> concurrentExceptions = new ArrayList<>();
+    private final Iterable<ExceptionHistoryEntry> concurrentExceptions;
```

Comment:
Shall we make this a `Set`? This is not a strong argument but `Iterable` feels as if this class does not really own the `ExceptionHistoryEntries`, whereas if it is a `Set` or any other `Collection`, then it feels a bit different.